packages/devp2p/src/rlpx/rlpx.ts

```diff
@@ -82,8 +82,10 @@ export class RLPx extends EventEmitter {
         if (this._peersLRU.has(peer.id!.toString('hex'))) return
         this._peersLRU.set(peer.id!.toString('hex'), true)
 
-        if (this._getOpenSlots() > 0) return this._connectToPeer(peer)
-        this._peersQueue.push({ peer: peer, ts: 0 }) // save to queue
+        if (this._getOpenSlots() > 0) {
+          return this._connectToPeer(peer)
+        }
+        this._getOpenSlots() > 0 && this._peersQueue.push({ peer: peer, ts: 0 }) // save to queue
```

Comment:
i'm a little confused by this code section, so if open slots > 0, we connect to the peer, but then this line would never get executed if it's guarded with the same condition? i assume this is then meant we have no open slots then we should save them to the queue?

Reply:
would also prefer if you used the same `if` style, rather than if block above then && on the next line

Reply:
It doesn't make sense to me either -- but changing that line does change the outcome.  This is kind of a bandaid.  I'll dig into it more.